crates/utils/src/config.rs

```diff
@@ -17,18 +17,29 @@ pub const DEFAULT_BLOCK_PRODUCER_PORT: u16 = 48046;
 pub const DEFAULT_STORE_PORT: u16 = 28943;
 pub const DEFAULT_FAUCET_SERVER_PORT: u16 = 8080;
 
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize, Default)]
+pub enum Protocol {
+    #[default] Http,
+    Https,
+}
 /// The `(host, port)` pair for the server's listening socket.
 #[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
 pub struct Endpoint {
     /// Host used by the store.
     pub host: String,
     /// Port number used by the store.
     pub port: u16,
+    /// Protocol type: http or https.
+    pub protocol: Protocol,
```

Comment:
Required so that the config deserialization gets the default value is missing.

```suggestion
    /// Protocol type: http or https.
    #[serde(default)]
    pub protocol: Protocol,
```